libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -652,6 +659,16 @@ bool RangeFinder::has_orientation(enum Rotation orientation) const
 // find first range finder instance with the specified orientation
 AP_RangeFinder_Backend *RangeFinder::find_instance(enum Rotation orientation) const
 {
+    // first check if user wants a particular instance to be used by default (For example, lua script backend)
+    for (uint8_t i=0; i<num_instances; i++) {
+        AP_RangeFinder_Backend *backend = get_backend(i);
+        if (backend != nullptr &&
+            backend->orientation() == orientation &&
+            backend->primary_sensor()) {
```

Comment:
We could also change to a "use" method. Like airspeed and compass. Rather than overriding with the scripting backend you would instead set to not use.

Reply:
Yes I think that adds moe functionality too. Done that